idp/idputil/idputil.go

```diff
@@ -254,3 +269,308 @@ func CookiePathRelativeToLocation(cookiePath, location string, skipLocation bool
 	}
 	return u.Path + cookiePath
 }
+
+// JsonResponse mashals the provided to JSON and writes it
+// to the provided response writer.
+func JsonResponse(w http.ResponseWriter, data interface{}, c int) {
+	dj, err := json.Marshal(data)
+	if err != nil {
+		http.Error(w, "failed to create a json response", http.StatusInternalServerError)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(c)
+	fmt.Fprintf(w, "%s", dj)
+}
+
+// WebAuthnUserFromIdentity converts the provided store Identity to
+// a type that implements the webauthn.User interface.
+func WebAuthnUserFromIdentity(id *store.Identity) webauthn.User {
+	return &webauthnUser{
+		Identity: id,
+	}
+}
+
+type webauthnUser struct {
+	*store.Identity
+}
+
+// WebAuthnID implements the webauthn.User interface.
+func (u *webauthnUser) WebAuthnID() []byte {
+	return []byte(u.Identity.Username)
```

Comment:
depending on what this is used for the ProviderID might be a better choice.

Reply:
done